backend/prompt_studio/prompt_studio_core_v2/serializers.py

```diff
@@ -44,48 +44,70 @@ class Meta:
 
     def to_representation(self, instance):  # type: ignore
         data = super().to_representation(instance)
+        default_profile = None
+
+        # Fetch summarize LLM profile
         try:
-            profile_manager = ProfileManager.objects.get(
+            summarize_profile = ProfileManager.objects.get(
                 prompt_studio_tool=instance, is_summarize_llm=True
             )
-            data[TSKeys.SUMMARIZE_LLM_PROFILE] = profile_manager.profile_id
+            data[TSKeys.SUMMARIZE_LLM_PROFILE] = summarize_profile.profile_id
         except ObjectDoesNotExist:
             logger.info(
-                "Summarize LLM profile doesnt exist for prompt tool %s",
+                "Summarize LLM profile doesn't exist for prompt tool %s",
                 str(instance.tool_id),
             )
+
+        # Fetch default LLM profile
         try:
-            profile_manager = ProfileManager.get_default_llm_profile(instance)
-            data[TSKeys.DEFAULT_PROFILE] = profile_manager.profile_id
+            default_profile = ProfileManager.get_default_llm_profile(instance)
+            data[TSKeys.DEFAULT_PROFILE] = default_profile.profile_id
         except DefaultProfileError:
             logger.warning(
-                "Default LLM profile doesnt exist for prompt tool %s",
+                "Default LLM profile doesn't exist for prompt tool %s",
                 str(instance.tool_id),
             )
-        prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.filter(
+
+        # Fetch prompt instances
+        prompt_instances = ToolStudioPrompt.objects.filter(
             tool_id=data.get(TSKeys.TOOL_ID)
         ).order_by("sequence_number")
-        data[TSKeys.PROMPTS] = []
+
+        if not prompt_instances.exists():
+            data[TSKeys.PROMPTS] = []
+            return data
+
+        # Process prompt instances
         output: list[Any] = []
-        # Appending prompt instances of the tool for FE Processing
-        if prompt_instance.count() != 0:
-            for prompt in prompt_instance:
-                profile_manager_id = prompt.prompt_id
-                if instance.single_pass_extraction_mode:
-                    # use projects default profile
-                    profile_manager_id = profile_manager.profile_id
-                prompt_serializer = ToolStudioPromptSerializer(prompt)
+        for prompt in prompt_instances:
+            prompt_serializer = ToolStudioPromptSerializer(prompt)
+            serialized_data = prompt_serializer.data
+
+            # Determine coverage
+            coverage: list[Any] = []
+            profile_manager_id = prompt.profile_manager
+            if default_profile and instance.single_pass_extraction_mode:
```

Comment:
@jagadeeswaran-zipstack How will the behavior be if the default profile is not selected and single pass is enabled? Are we prompting anywhere to select default profile if single pass is enabled?